qiskit/chemistry/drivers/fcidumpd/parser.py

```diff
@@ -0,0 +1,227 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" FCIDump parser """
+
+import itertools
+import re
+import numpy as np
+
+from qiskit.chemistry import QiskitChemistryError
+
+
+def parse(fcidump: str) -> dict:
```

Comment:
Use `Dict[key type, value type]` instead of `dict`

Reply:
What should I put for `value type` in this case?
It ranges from `int`, `float` to `List[int]` and `List[float]`?

Reply:
`List[Union[int, float, List[int], List[float]]` or `List[Any]`